### PR TITLE
Config Serialization

### PR DESCRIPTION
**Pull request by saladoc**

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/6c6c2d30/index.html.

Here are some links to the pages that were modified:

- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/6c6c2d30/plugin/configuration/index.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/6c6c2d30/plugin/configuration/loaders.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/6c6c2d30/plugin/injection.html
- added https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/6c6c2d30/plugin/configuration/serialization.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/327#issuecomment-138599574), not to the direct url to the preview._

File: source/plugin/configuration/serialization.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+        @Setting(value="player")
+        private UUID playerUUID;
+        @Setting(comment="Number of diamonds mined")
+        private int diamonds;
+
+        [...]
+    }
+
+The above example can now be serialized and deserialized from config nodes without further registration. The ``@Setting`` annotations map a configuration node to the field annotated. It accepts two optional parameters, ``value`` and ``comment`. If the ``value`` parameter exists, it defines the name of the node the field will be saved in. If it is not present, the name of the field will be used instead. So in our above example, the annotation ensures that the contents of the field ``playerUUID`` are saved to the node "player". The ``diamonds`` field however will be saved under that exact name since its annotation only specifies a comment. That comment will be written to the config if the implementation supports commented configuration nodes, else it will be discarded.
```
Comment (Tzky) [inline source/plugin/configuration/serialization.rst]: add additional `` to ``comment`

**Comment by Tzky** [comment]
When looking at the prieview of serialization page:
codeblocks and note/tip/warning boxes seem to get shifted to the right. that shouldn't be this way, probably due to spacing of the blocks in the source.

**Comment by boformer** [comment]
Looks good.

Instead of ``@Setting(value="player")``, you can also use ``@Setting("player")``

**Pull request closed without merging**